synfig-studio/src/gui/main.cpp

```diff
@@ -129,11 +129,19 @@ int main(int argc, char **argv)
 	cout << "   " << _("synfig studio -- starting up application...") << endl << endl;
 
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
-	studio::App app(etl::dirname(binary_path), &argc, &argv);
+	
+	// studio::App app(etl::dirname(binary_path), &argc, &argv);
+	Glib::RefPtr<studio::App> app = studio::App::create();
```

Comment:
Why don't you pass the binary_path data to create() as it was when calling directly the constructor?

Why do you pass the argc/argv via lambda signal slot, but not as to app::run method, as in the "trivial example" of the docs?
```c++
auto application = ExampleApplication::create();

  // Start the application, showing the initial window,
  // and opening extra views for any files that it is asked to open,
  // for instance as a command-line parameter.
  // run() will return when the last window has been closed.
  return application->run(argc, argv);

```

Reply:
This part of the code is suggested by @ice0 so that is why I am sticking to this :) Kindly confirm it once if I have to change it.

Reply:
@DhairyaBahl 
It is bad practice to copy/paste the code without understanding how it works.
If you do not understand why this is implemented in one way and not in another, feel free to ask. 

@rodolforg 
If you use `run(argc, argv)`, you will need to implement `signal_handle_local_options()` and rewrite the parsing code, and do other things that need to be done, but my suggestion is to skip them for now, because there are only about two weeks of coding period is left, and we are moving very slowly.

Reply:
@ice0 Okay ! I will keep that in mind from next time 🙂

Reply:
So we use `studio::App::create(int& argc, char**& argv)` instead of just `studio::App::create()`.
The `create` method can call `String binary_path = synfig::get_binary_path(String(argv[0]));` itself, instead of `main()` entry point.

Reply:
I don't understand your idea. How do you plan to call the `init()` method?
`init()` should only be called once, `create()` (because it returns an `App` pointer), may be called multiple times.